apps/frontend/src/components/Spinner.tsx

```diff
@@ -0,0 +1,65 @@
+import CircularProgress from "@mui/material/CircularProgress";
+import { Grid, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+
+interface Props {
+  text: string;
+}
+
+export default function Spinner({ text }: Props) {
+  const [isAlive, setIsAlive] = useState<boolean>(true);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setIsAlive(false);
+    }, 40000);
+    return () => clearTimeout(timer);
+  }, []);
+
+  return (
+    <>
+      {isAlive ? (
+        <Grid
+          gap={2}
+          container
+          direction="column"
+          alignItems="center"
+          justifyContent="center"
+          sx={{ marginTop: "40px" }}
+        >
+          <Grid item>
+            <CircularProgress size="40px" />
+          </Grid>
+          <Grid item sx={{ textAlign: "center" }}>
+            <Typography variant="h5">Hang with us</Typography>
+            <Typography variant="h6">{text}</Typography>
+          </Grid>
```

Comment:
If you do decide to keep this component, this is the only difference between the 2 states, so you can just have conditionals here on the spinner and the text

Reply:
I tried it a few ways and I thought the way it is now was the simplest to read.